rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -117,7 +117,18 @@ rclcpp::utilities::init(int argc, char * argv[])
     // NOLINTNEXTLINE(runtime/arrays)
     char error_string[error_length];
 #ifndef _WIN32
-    strerror_r(errno, error_string, error_length);
+#ifdef _GNU_SOURCE
+    char * msg = strerror_r(errno, error_string, error_length);
+    if (msg != error_string) {
+      strncpy(error_string, msg, sizeof(error_string));
```

Comment:
The variable `error_length` already contains the value of `sizeof(error_string)`.

Same next line.